src/server/pfs/fuse/filesystem_test.go

```diff
@@ -26,695 +30,302 @@ func TestRootReadDir(t *testing.T) {
+func TestWriteAndRead(t *testing.T) {
+	lion.SetLevel(lion.LevelDebug)
+	if testing.Short() {
+		t.Skip("Skipped because of short mode")
 	}
 
-	ready := make(chan bool)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		if err := mounter.Mount(mountpoint, nil, nil, ready); err != nil {
-			t.Errorf("mount and serve: %v", err)
-		}
-	}()
-
-	<-ready
-	defer func() {
-		if err := mounter.Unmount(mountpoint); err != nil {
-			t.Errorf("unmount: %v", err)
-		}
-	}()
+	testFuse(t, func(apiClient pfsclient.APIClient, mountpoint string) {
+		repoName := "foo"
+		require.NoError(t, pfsclient.CreateRepo(apiClient, repoName))
+		commit, err := pfsclient.StartCommit(apiClient, repoName, "", "")
+		require.NoError(t, err)
+		greeting := "Hello, world\n"
+		filePath := filepath.Join(mountpoint, repoName, commit.ID, "greeting")
+		require.NoError(t, ioutil.WriteFile(filePath, []byte(greeting), 0644))
+		_, err = ioutil.ReadFile(filePath)
+		// errors because the commit is unfinished
+		require.YesError(t, err)
+		require.NoError(t, pfsclient.FinishCommit(apiClient, repoName, commit.ID))
+		data, err := ioutil.ReadFile(filePath)
+		require.NoError(t, err)
+		require.Equal(t, []byte(greeting), data)
+	})
+}
 
-	const (
-		repoName = "foo"
-	)
-	if err := pfsclient.CreateRepo(apiClient, repoName); err != nil {
-		t.Fatalf("CreateRepo: %v", err)
-	}
-	commit, err := pfsclient.StartCommit(apiClient, repoName, "", "")
-	if err != nil {
-		t.Fatalf("StartCommit: %v", err)
+func TestBigWrite(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipped because of short mode")
 	}
-	t.Logf("open commit %v", commit.ID)
 
-	const (
-		greetingName = "greeting"
-		greeting     = "Hello, world\n"
-		greetingPerm = 0644
-	)
-	if err := ioutil.WriteFile(filepath.Join(mountpoint, repoName, commit.ID, greetingName), []byte(greeting), greetingPerm); err != nil {
-		t.Fatalf("WriteFile: %v", err)
-	}
-	const (
-		scriptName = "script"
-		script     = "#!/bin/sh\necho foo\n"
-		scriptPerm = 0750
-	)
-	if err := ioutil.WriteFile(filepath.Join(mountpoint, repoName, commit.ID, scriptName), []byte(script), scriptPerm); err != nil {
-		t.Fatalf("WriteFile: %v", err)
-	}
+	testFuse(t, func(apiClient pfsclient.APIClient, mountpoint string) {
+		repo := "test"
+		require.NoError(t, pfsclient.CreateRepo(apiClient, repo))
+		commit, err := pfsclient.StartCommit(apiClient, repo, "", "")
+		require.NoError(t, err)
+		path := filepath.Join(mountpoint, repo, commit.ID, "file1")
+		stdin := strings.NewReader(fmt.Sprintf("yes | tr -d '\\n' | head -c 1000000 > %s", path))
+		require.NoError(t, pkgexec.RunStdin(stdin, "sh"))
+		require.NoError(t, pfsclient.FinishCommit(apiClient, repo, commit.ID))
+		data, err := ioutil.ReadFile(path)
+		require.NoError(t, err)
+		require.Equal(t, bytes.Repeat([]byte{'y'}, 1000000), data)
+	})
+}
 
-	if err := pfsclient.FinishCommit(apiClient, repoName, commit.ID); err != nil {
-		t.Fatalf("FinishCommit: %v", err)
+func Test296(t *testing.T) {
```

Comment:
Yea, I like naming it 'TestIssue296' - thats probably enough since github
URLs are semantic (easy to copy paste the number)

On Tue, Apr 19, 2016 at 6:43 PM, Joe Doliner <notifications@github.com>
wrote:

> In src/server/pfs/fuse/filesystem_test.go
> <https://github.com/pachyderm/pachyderm/pull/275#discussion_r60340088>:
>
> >
> > -	if err := pfsclient.FinishCommit(apiClient, repoName, commit.ID); err != nil {
> > -		t.Fatalf("FinishCommit: %v", err)
> > +func Test296(t *testing.T) {
>
> Oh this test is recreating the situation described in #296
> <https://github.com/pachyderm/pachyderm/issues/296> . I kind of like
> adding regression tests for specific issues, what about TestIssue296, and
> maybe a comment with a link to the original issue?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/pachyderm/pachyderm/pull/275/files/1aa62f715c1c21e9ebd91ba885650dc6950cf739#r60340088>
>